pkg/plugins/generic/generic_plugin.go

```diff
@@ -233,6 +238,49 @@ func isCommandNotFound(err error) bool {
 	return false
 }
 
+// addToDesiredKernelArgs Should be called to queue a kernel arg to be added to the node.
+func (p *GenericPlugin) addToDesiredKernelArgs(karg string) {
+	if _, ok := p.DesiredKernelArgs[karg]; !ok {
+		glog.Infof("generic-plugin addToDesiredKernelArgs(): Adding %s to desired kernel arg", karg)
+		// element "uint" is a counter of number of attempts to set the kernel argument
+		p.DesiredKernelArgs[karg] = 0
+	}
+}
+
+// syncDesiredKernelArgs Should be called to set all the kernel arguments. Returns bool if node update is needed.
+func (p *GenericPlugin) syncDesiredKernelArgs() (bool, error) {
+	needReboot := false
+	if len(p.DesiredKernelArgs) != 0 {
+		kargs, err := utils.GetCurrentKernelArgs(false)
+		if err != nil {
+			return false, err
+		}
+		for desiredKarg, attempts := range p.DesiredKernelArgs {
+			set := utils.IsKernelArgsSet(kargs, desiredKarg)
+			if !set {
+				if attempts > 0 {
+					glog.Errorf("generic-plugin syncDesiredKernelArgs(): failed to set kernel arg %s with attempts %d", desiredKarg, attempts)
+				}
+				// There is a case when we try to set the kernel argument here, the daemon could decide to not reboot because
+				// the daemon encountered a potentially one-time error. However we always want to make sure that the kernel
+				// argument is set once the daemon goes through node state sync again.
+				update, err := setKernelArg(desiredKarg)
+				if err != nil {
+					glog.Errorf("generic-plugin syncDesiredKernelArgs(): fail to set kernel arg %s: %v", desiredKarg, err)
+					return false, err
+				}
+				if update {
+					needReboot = true
+					glog.V(2).Infof("generic-plugin syncDesiredKernelArgs(): need reboot for setting kernel arg %s", desiredKarg)
+				}
+				// Update the number of attempts we tried to set the kernel argument.
+				p.DesiredKernelArgs[desiredKarg]++
```

Comment:
I'm somewhat confused again by tracking attempts. Can you provide an example of when it will be > 1?

Reply:
Correct me if I'm wrong, but I don't think it will be more than 1 at any point. It seems to me that a bool is a better choice here.

Reply:
Reading through the code, I believe that is correct. I just wanted a way to track it in case it ever happens. But bool should suffice here. made the changes.